graphics/libfive/Portfile

```diff
@@ -0,0 +1,165 @@
+    set p python${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set i [lsearch -exact ${pythons_ports} ${p}]
+    set c [lreplace ${pythons_ports} ${i} ${i}]
+
+    variant ${p} description "Build bindings for Python ${v}" conflicts {*}${c} "
+        depends_lib-append      port:${p}
+        configure.args-replace  -DBUILD_PYTHON_BINDINGS=OFF -DBUILD_PYTHON_BINDINGS=ON
+        pre-patch {
+            # Patch the CMake project to use the selected version of Python
+            reinplace \"s|Python3 COMPONENTS|Python3 ${v} EXACT COMPONENTS|g\" ${worksrcpath}/CMakeLists.txt
+        }
+    "
+}
+
+# Studio app:
+
+variant studio requires guile description {Build GUI app} {
+    PortGroup               qt5 1.0
+
+    qt5.min_version         5.12
+
+    configure.args-replace  -DBUILD_STUDIO_APP=OFF -DBUILD_STUDIO_APP=ON
+
+    # The Studio app is under GPL-2+
+    if { ! ("GPL-2+" in $license) } {
+        license-append GPL-2+
+    }
+
+    # The Studio app only uses Qt, but does not use OpenSSL directly.
+    # Qt's licenses include OpenSSLException.
+    license_noconflict      openssl
+
+    depends_build-append    port:librsvg
+
+    post-build {
+        # Create app icon
+
+        set svg    ${worksrcpath}/studio/deploy/icon/icon.svg
+
+        file mkdir ${build.dir}/icon.iconset
+
+        foreach res {16 32 128 256 512} {
+            set hres [expr 2*${res}]
+            system -W ${worksrcpath} "${prefix}/bin/rsvg-convert -w ${res} ${svg}  > ${build.dir}/icon.iconset/icon_${res}x${res}.png"
+            system -W ${worksrcpath} "${prefix}/bin/rsvg-convert -w ${hres} ${svg} > ${build.dir}/icon.iconset/icon_${res}x${res}@2x.png"
+        }
+
+        file mkdir ${build.dir}/studio/Studio.app/Contents/Resources
+
+        system -W ${worksrcpath} "/usr/bin/iconutil -c icns -o ${build.dir}/studio/Studio.app/Contents/Resources/studio.icns ${build.dir}/icon.iconset"
+
+        # Include actual version into Info.plist
+        reinplace "s|0\.0\.0|${version}|g" ${build.dir}/studio/Studio.app/Contents/Info.plist
+    }
+
+    post-destroot {
+        move ${build.dir}/studio/Studio.app ${destroot}${applications_dir}
+
+        set examplesdir ${prefix}/share/${name}/examples
+
+        xinstall -d ${destroot}${examplesdir}
+        xinstall -m 0644 {*}[glob ${worksrcpath}/studio/examples/*.io] ${destroot}${examplesdir}
+
+        notes-append "${name} studio examples were installed into ${examplesdir}."
+    }
+}
+
+# Testing:
+
+# As of version 20210424 there may be test failures for REQUIRE( a.count() == 2 )
+# See https://github.com/libfive/libfive/issues/424
+
+test.run            yes
+test.cmd            ${build.dir}/libfive/test/libfive-test
+test.target
+
+# Default variants:
+
+set selected_python python39
+foreach s ${pythons_suffixes} {
+    if {[variant_isset python${s}]} {
+        set selected_python python${s}
+    }
+}
+default_variants    +${selected_python}
+
+default_variants    +studio +guile
```

Comment:
this doesn't look right to me. You probably want to have them either all three variants set at the same time (preferred), otherwise you'll need to use `default_variants-append` not to overwrite the first one you set

Reply:
Are you sure this is necessary? Other ports use `default_variant` (and not `default_variant-append`) this way as well, including `boost` from where I borrowed this code.  It does appear to work fine, without overwriting previous default variants.

Nevertheless, I changed it to use a single `default_variants` line now.

For reference, your comment referred to this code:

```
default_variants    +${selected_python}

default_variants    +studio +guile
```

Reply:
I am actually not sure, it might be that `default_variants` behaves differently than for example `depends_lib`. I was just assuming that one would overwrite the previous default variants in the way you did it here (but you could certainly test that). There are many ports that also do `default_variants-append` so that's why I was thinking about it.

Reply:
Since this is resolved, can we merge this?

Reply:
Renee brought up a good point: While `default_variants` may provide the same result as `default_variants-append`, the latter is highly preferable. If nothing else, it provides clear semantics regarding what wants to be done.

The change to a single `default_variants` statement is fine, and I merged this PR. But going forward, we should ensure folks use `default_variants-append` instead, if possible.

Reply:
Thanks for the clarification!